core/class.go

```diff
@@ -52,3 +64,49 @@ func flatten(entryPoints []EntryPoint) []*felt.Felt {
 	}
 	return result
 }
+
+// Cairo1Class unambiguously defines a [Contract]'s semantics.
+type Cairo1Class struct {
+	Abi         string
+	AbiHash     *felt.Felt
+	EntryPoints struct {
+		Constructor []SierraEntryPoint
+		External    []SierraEntryPoint
+		L1Handler   []SierraEntryPoint
+	}
+	Program         []*felt.Felt
+	ProgramHash     *felt.Felt
+	SemanticVersion string
+}
+
+type SierraEntryPoint struct {
+	Index    uint64
+	Selector *felt.Felt
+}
+
+func (c *Cairo1Class) Version() uint64 {
+	return 1
+}
+
+func (c *Cairo1Class) Hash() *felt.Felt {
+	contractClassVersion := fmt.Sprintf("CONTRACT_CLASS_V%s", c.SemanticVersion)
+	return crypto.PoseidonArray(
+		new(felt.Felt).SetBytes([]byte(contractClassVersion)),
```

Comment:
I believe we can [avoid](https://github.com/uber-go/guide/blob/master/style.md#prefer-strconv-over-fmt) an allocation from `fmt.Sprintf`. 
```suggestion
	return crypto.PoseidonArray(
		new(felt.Felt).SetBytes([]byte("CONTRACT_CLASS_V"+c.SemanticVersion)),
```

Reply:
Will apply since it is one less line of code, but I believe this will still allocate since strings are immutable in go.